File: web-ui/src/models/remote-connectors-settings/serial-remote-connector-settings-model.ts
import { SerialRemoteConnectorBaudsModel } from './serial-remote-connector-bauds-model';
import { SerialRemoteConnectorBytesizesModel } from './serial-remote-connector-bytesizes-model';
import { SerialRemoteConnectorParitiesModel } from './serial-remote-connector-parities-model';
import { SerialRemoteConnectorStopbitsModel } from './serial-remote-connector-stopbits-model';


export type SerialRemoteConnectorSettingsModel = {
    port: string;

    baud: SerialRemoteConnectorBaudsModel;

    timeout: number;

    parity: SerialRemoteConnectorParitiesModel;

    stopbits: SerialRemoteConnectorStopbitsModel;

    bytesize: SerialRemoteConnectorBytesizesModel;
};

File: web-api/src/models/remote_connector/serial_remote_connector_settings_model.py
from models.abstracts.app_base_model import AppBaseModel
from models.remote_connector.serial_remote_connector_parities_model import SerialRemoteConnectorParitiesModel
from models.remote_connector.serial_remote_connector_bauds_models import SerialRemoteConnectorBaudsModel
from models.remote_connector.serial_remote_connector_bytesize_models import SerialRemoteConnectorBytesizeModel
from models.remote_connector.serial_remote_connector_stopbits_model import SerialRemoteConnectorStopbitsModel


class SerialRemoteConnectorSettingsModel(AppBaseModel):

    port: str

    baud: SerialRemoteConnectorBaudsModel

    timeout: int

    parity: SerialRemoteConnectorParitiesModel

    stopbits: SerialRemoteConnectorStopbitsModel

    bytesize: SerialRemoteConnectorBytesizeModel